src/main/groovy/au/org/ala/ws/security/authenticator/AlaApiKeyAuthenticator.groovy

```diff
@@ -0,0 +1,121 @@
+package au.org.ala.ws.security.authenticator
+
+import au.org.ala.ws.security.profile.AlaApiUserProfile
+import com.nimbusds.common.contenttype.ContentType
+import com.nimbusds.jose.shaded.json.JSONObject
+import com.nimbusds.oauth2.sdk.ParseException
+import com.nimbusds.oauth2.sdk.http.HTTPRequest
+import com.nimbusds.oauth2.sdk.http.HTTPResponse
+import org.pac4j.core.context.WebContext
+import org.pac4j.core.context.session.SessionStore
+import org.pac4j.core.credentials.Credentials
+import org.pac4j.core.credentials.TokenCredentials
+import org.pac4j.core.credentials.authenticator.Authenticator
+import org.pac4j.core.exception.CredentialsException
+import org.pac4j.core.util.CommonHelper
+import org.pac4j.core.util.InitializableObject
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.stereotype.Component
+
+@Component
+@ConditionalOnProperty([ 'security.apikey.enabled', 'security.jwt.fallback-to-legacy-behaviour' ])
+class AlaApiKeyAuthenticator extends InitializableObject implements Authenticator {
+
+    @Value('security.apikey.check.serviceUrl')
+    String apiKeyUri
+
+    @Value('security.apikey.userdetails.serviceUrl')
+    String userDetailsUri
+
+    @Override
+    protected void internalInit(boolean forceReinit) {
+        CommonHelper.assertNotBlank("apiKeyUri", apiKeyUri);
+        CommonHelper.assertNotBlank("userDetailsUri", userDetailsUri);
+    }
+
+    @Override
+    void validate(Credentials credentials, WebContext context, SessionStore sessionStore) {
+
+        init()
+
+        TokenCredentials alaApiKeyCredentials = (TokenCredentials) credentials
+
+        AlaApiUserProfile alaApiUserProfile = fetchUserProfile(alaApiKeyCredentials.token)
+
+        if (alaApiUserProfile.activated && !alaApiUserProfile.locked) {
+
+            alaApiKeyCredentials.userProfile = alaApiUserProfile
+        }
+    }
+
+    AlaApiUserProfile fetchUserProfile(String apiKey) {
+
+        AlaApiUserProfile alaApiUserProfile = new AlaApiUserProfile()
+
+        HTTPRequest apiKeyRequest = new HTTPRequest(HTTPRequest.Method.GET, URI.create("${apiKeyUri}/${apiKey}"))
```

Comment:
IMHO These HTTP calls should be extracted to eg a retrofit interface dependency so that they can be unit tested without spinning up an HTTP server.